src/thor/dijkstras.cc

```diff
@@ -183,7 +184,7 @@ void Dijkstras::ExpandForward(GraphReader& graphreader,
     uint32_t idx = bdedgelabels_.size();
     *es = {EdgeSet::kTemporary, idx};
     bdedgelabels_.emplace_back(pred_idx, edgeid, oppedgeid, directededge, newcost, newcost.cost, 0.0f,
-                               mode_, transition_cost, false, restriction_idx);
+                               mode_, path_dist, transition_cost, false, restriction_idx);
```

Comment:
I did need to add an overload on the `BDEdgeLabel`s constructor to support this, see below..

Reply:
there is an overload that does take this info already:

https://github.com/valhalla/valhalla/blob/master/valhalla/sif/edgelabel.h#L471

i think that one will work for you. i know its pretty annoying to figure out what is what especially with emplace_back

Reply:
yep you're right..

Reply:
done